src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java

```diff
@@ -53,6 +55,14 @@ public boolean isCaseTypeUsingTTL(@NonNull CaseTypeDefinition caseTypeDefinition
             .stream().anyMatch(caseFieldDefinition -> TTL_CASE_FIELD_ID.equals(caseFieldDefinition.getId()));
     }
 
+    private Boolean isNullifyByDefault(@NonNull CaseEventDefinition caseEventDefinition) {
+        List<CaseEventFieldDefinition> caseEventFieldDefinitions =
+            Optional.ofNullable(caseEventDefinition.getCaseFields()).orElse(Collections.emptyList())
+                .stream().filter(cefDefinition -> TTL_CASE_FIELD_ID.equals(cefDefinition.getCaseFieldId()))
+                .filter(cefDefinition -> cefDefinition.getNullifyByDefault() != null).toList();
+        return !caseEventFieldDefinitions.isEmpty() && caseEventFieldDefinitions.getFirst().getNullifyByDefault();
```

Comment:
Could probably be simplified a bit and easier to read:
```suggestion
        return caseEventDefinition.getCaseEventField(TTL_CASE_FIELD_ID)
            .map(CaseEventFieldDefinition::getNullifyByDefault)
            .orElse(false);
```